deepspeed/runtime/engine.py

```diff
@@ -1163,6 +1176,10 @@ def forward(self, *inputs, **kwargs):
         return loss
 
     def allreduce_gradients(self, bucket_size=MEMORY_OPT_ALLREDUCE_SIZE):
+        # Pass (PP) gas boundary flag to optimizer (required for zero)
+        self.optimizer.is_gradient_accumulation_boundary = self.is_gradient_accumulation_boundary(
```

Comment:
Is `self.optimizer` guaranteed to have `is_gradient_accumulation_boundary` attribute?

Reply:
We traced the break to this line. Crash went away after commenting this line.